src/cosalib/cli.py

```diff
@@ -0,0 +1,119 @@
+# NOTE: PYTHONUNBUFFERED is set in cmdlib.sh for unbuffered output
+# pylint: disable=C0103
+
+import argparse
+import logging as log
+import os
+
+#: All valid logger levels
+LOG_LEVELS = ('warn', 'error', 'debug', 'info')
+
+
+class Cli:
+    """
+    Abstraction for executing commands from the cli.
+    """
+
+    def __init__(self, **kwargs):
+        """
+        Initializes the Cli instance.
+
+        :param kwargs: All keyword arguments which will pass to ArgumentParser
+        :type kwargs: dict
+        """
+        self.parser = argparse.ArgumentParser(**kwargs)
+        self.add_env_arg(
+            'COSA_LOG_LEVEL', 'info', '--log-level',
+            choices=LOG_LEVELS, help='Set the log level')
+
+    def add_env_arg(self, env_var, default, *args, **kwargs):
+        """
+        Adds an argument that can also pull it's value from an
+        environment variable.
+
+        :param env_var: The environment variable to look at
+        :type env_var: str
+        :env default: The default for the argument
+        :type default: any
+        :param args: Non keyword arguments to pass to add_argument
+        :type args: list
+        :param kwargs: Keyword arguments to pass to add_argument
+        :type kwargs: dict
+        """
+        if kwargs.get('help') is None:
+            kwargs['help'] = ''
+        kwargs['help'] = kwargs['help'] + ' (Env: {})'.format(env_var)
+        self.parser.add_argument(
+            *args, default=os.environ.get(env_var, default), **kwargs)
+
+    def parse_args(self):
+        """
+        Parses the arguments passed in, verifies inputs, sets the logger,
+        and returns the arguments.
+
+        :returns: The parsed arguments
+        :rtype: argparse.Namepsace
+        """
+        args = self.parser.parse_args()
+        self._set_logger(args.log_level)
+        return args
+
+    def _set_logger(self, level):
+        """
+        Set the log level
+
+        :param level: set the log level
+        :type level: str
+        """
+        sl = log.DEBUG
```

Comment:
Any reason not to change the `LOG_LEVELS` variable into a dictionary to allow you to do a simpler lookup? For example you could do something like so:

```python
LOG_LEVELS = {"warn": log.WARN, "error": log.ERROR, ...}

def _set_logger(self, level):
  log.basicConfig(
    format='[%(asctime)s  %(levelname)s]: %(message)s',
    level=LOG_LEVELS.get(level, log.DEBUG)
```